aws/structure.go

```diff
@@ -1451,14 +1451,30 @@ func expandApiGatewayStageKeyOperations(d *schema.ResourceData) []*apigateway.Pa
 	return operations
 }
 
-func expandCloudWachLogMetricTransformations(m map[string]interface{}) []*cloudwatchlogs.MetricTransformation {
+func expandCloudWachLogMetricTransformations(m map[string]interface{}) ([]*cloudwatchlogs.MetricTransformation, error) {
 	transformation := cloudwatchlogs.MetricTransformation{
 		MetricName:      aws.String(m["name"].(string)),
 		MetricNamespace: aws.String(m["namespace"].(string)),
 		MetricValue:     aws.String(m["value"].(string)),
 	}
 
-	return []*cloudwatchlogs.MetricTransformation{&transformation}
+	if m["default_value"] != "" {
+		bound := m["default_value"]
+		switch bound := bound.(type) {
+		case string:
+			f, err := strconv.ParseFloat(bound, 64)
+			if err != nil {
+				return nil, fmt.Errorf(
+					"default_value must be a float value represented as a string")
```

Comment:
Do you think it would be worth to make a validation function for that?

Reply:
@Ninir I changed to use the built in schema.TypeFloat to avoid the need for parsing